File: AEDS2/Tp1/q1/Palindromo.java
import java.util.*;

//checa se a string é palindromo
class Palindromo {
    public static boolean isPalindromo ( String s ) {
        boolean result = true;
        for ( int y = 0; y < s.length()/2 && result; y++ ) {
            if ( s.charAt(y) != s.charAt(s.length() - y - 1) ) {
                result = false;
            }
        }
        return result;
    }

    //main que le entrada e retorna se é palindromo
    public static void main (String args[]) {
        String s;
        s = MyIO.readLine();
        while ( ! (s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M') ) {
            if ( isPalindromo(s) ) {
                System.out.println("SIM");
            } else {
                System.out.println("NAO");
            }
            s = MyIO.readLine();
        }
    }
}

File: AEDS2/Tp1/q2/Palindromo.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

//checa se a string é palindromo
bool isPalindromo ( char s[] ) {
        bool result = true;
        int l;
        l = strlen(s);
        for (int y = 0; y < l/2 && result; y++) {
            if ( s[y] != s[l-y-1] ) {
                result = false;
            }
        }
        return result;
    }

//checa se a string é "FIM"
bool isFim (char s[]) {
    bool result = false;
    if (s[0] == 'F' && s[1] == 'I' && s[2] == 'M') { result = 1; } 
    return result;
}

//main que le entrada e retorna se é palindromo
int main () {
    char s[1000];
    scanf( "%[^\n]", s );
    while ( ! isFim(s) ) {
        if (isPalindromo(s)) {
            printf("SIM\n");
        } else {
            printf("NAO\n");
        }
        scanf( "%[^\n]", s );
    }

    return 0;
}